builder/vsphere/driver/vm.go

```diff
@@ -351,52 +351,56 @@ func (vm *VirtualMachineDriver) Clone(ctx context.Context, config *CloneConfig)
 
 	virtualDisks := devices.SelectByType((*types.VirtualDisk)(nil))
 	virtualControllers := devices.SelectByType((*types.VirtualController)(nil))
+	virtualNetworkAdapters := devices.SelectByType((*types.VirtualEthernetCard)(nil))
 
 	// Use existing devices to avoid overlapping configuration
 	existingDevices := object.VirtualDeviceList{}
 	existingDevices = append(existingDevices, virtualDisks...)
 	existingDevices = append(existingDevices, virtualControllers...)
+	existingDevices = append(existingDevices, virtualNetworkAdapters...)
 
 	storageConfigSpec, err := config.StorageConfig.AddStorageDevices(existingDevices)
 	if err != nil {
 		return nil, fmt.Errorf("failed to add storage devices: %s", err.Error())
 	}
 	configSpec.DeviceChange = append(configSpec.DeviceChange, storageConfigSpec...)
 
-	if config.Network != "" {
-		net, err := vm.driver.FindNetwork(config.Network)
-		if err != nil {
-			return nil, fmt.Errorf("Error finding network: %s", err)
-		}
-		backing, err := net.network.EthernetCardBackingInfo(ctx)
-		if err != nil {
-			return nil, fmt.Errorf("Error finding ethernet card backing info: %s", err)
-		}
-
-		devices, err := vm.vm.Device(ctx)
-		if err != nil {
-			return nil, fmt.Errorf("Error finding vm devices: %s", err)
+	networkNames := splitTrim(config.Network)
+	if len(networkNames) > 0 {
+		if len(networkNames) > len(virtualNetworkAdapters) {
+			return nil, fmt.Errorf("%d network names defined but only %d network adapters exist", len(networkNames), len(virtualNetworkAdapters))
```

Comment:
Nice way to validate that list of provided network names match the count of actual adapters.